### PR TITLE
feat: user-service

### PR DESCRIPTION
**Pull request by Fl1s**

### CODE REVIEW

**Review by neojelll — CHANGES_REQUESTED** [review]
мне не понравилось переделывай

File: .env.example
Comment (neojelll) [inline .env.example:1]: Why deleted .env.example???

File: apps/user-service/src/schemas/user.schema.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
+import { Prop, Schema, SchemaFactory } from '@nestjs/mongoose';
+import { Document } from 'mongoose';
+import { v4 as uuidv4 } from 'uuid';
+@Schema({ timestamps: true })
```
Comment (neojelll) [inline apps/user-service/src/schemas/user.schema.ts:4]: new line?

File: apps/user-service/src/user-service.controller.ts [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+  ) {
+    return await this.userService.getUserByUserId(userId);
+  }
+
+  @MessagePattern('user.update')
+  async handleUpdateUser(
+    @Payload() payload: { userId: string; data: UserUpdateDto },
+    @Ctx() context: KafkaContext,
+  ) {
+    return await this.userService.updateUser(payload.userId, payload.data);
```
Comment (neojelll) [inline apps/user-service/src/user-service.controller.ts:44]: 1. you write `@Ctx() context: KafkaContext` but not use

2. you write static topic name please get it from .env file

File: apps/user-service/src/user-service.module.ts [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+            clientId: 'user-service',
+            brokers: [process.env.MESSAGE_BROKER_URL || 'localhost:9094'],
+          },
+          consumer: {
+            groupId: 'user-service-consumer',
+          },
+        },
+      },
+    ]),
+  ],
```
Comment (neojelll) [inline apps/user-service/src/user-service.module.ts:29]: 1. KAFKA_SERVICE is so-so name

File: apps/user-service/test/app.e2e-spec.ts [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+    await app.init();
+  });
+
+  it('/ (GET)', () => {
+    return request(app.getHttpServer())
+      .get('/')
+      .expect(200)
+      .expect('Hello World!');
+  });
+});
```
Comment (neojelll) [inline apps/user-service/test/app.e2e-spec.ts:24]: i think there is no need to keep a folder with tests if there are none

**Review by neojelll — CHANGES_REQUESTED** [review]
i think it`s still worth it using PostgreSQL for store users

File: apps/user-service/src/schemas/user.schema.ts (hunk 1 above)
Comment (Fl1s) [reply apps/user-service/src/schemas/user.schema.ts:4]: Yes, timestamp's used to generate createdAt/updatedAt values

File: apps/user-service/src/user-service.module.ts (hunk 3 above)
Comment (Fl1s) [reply apps/user-service/src/user-service.module.ts:29]: USER_SERVICE???

Comment (Fl1s) [reply apps/user-service/src/user-service.module.ts:29]: or USERS_MICROSERVICE????

File: apps/user-service/test/app.e2e-spec.ts (hunk 4 above)
Comment (Fl1s) [reply apps/user-service/test/app.e2e-spec.ts:24]: Np

File: .env.example
Comment (Fl1s) [reply .env.example:1]: added back

File: README.md [hunk 5]
```diff
@@ -1,43 +1,59 @@
...
+	  <a href="https://github.com/neojelll/DocuTide" target="_blank">DocuTide</a> - The Ultimate Service for Effortless Documentation Creation!
+	</b> 
 </p>
 
-<p align="center">
-	<img alt="GitHub Release" src="https://img.shields.io/github/v/release/neojelll/DocuTide?include_prereleases&display_name=release&style=flat">
+<p align="center"> 
+  <img alt="GitHub Release" src="https://img.shields.io/github/v/release/neojelll/DocuTide?include_prereleases&display_name=release&style=flat">
 	<img alt="Static Badge" src="https://img.shields.io/badge/TypeScript-5.5-blue?style=flat">
 	<img alt="Coveralls" src="https://img.shields.io/coverallsCoverage/github/neojelll/DocuTide?style=flat">
```
[the remark this answers is not in the document]
Comment (Fl1s) [reply README.md]: idk, maybe I just flabbergasted by hotkeys and suddenly writed smth into readme

File: docker/docker-compose.yml
Comment (neojelll) [inline docker/docker-compose.yml:1]: but better so no make

File: apps/user-service/src/schemas/user.schema.ts (hunk 1 above)
Comment (neojelll) [reply apps/user-service/src/schemas/user.schema.ts:4]:
```suggestion
import { v4 as uuidv4 } from 'uuid';
@Schema({ timestamps: true })
```

why here not new line?

**Review by neojelll — DISMISSED** [review]

File: docker/docker-compose.yml
Comment (Fl1s) [reply docker/docker-compose.yml:1]: ???

**Review by neojelll — APPROVED** [review]
good job bro
